python/pyspark/sql/types.py

```diff
@@ -1397,6 +1397,8 @@ def _create_row_inbound_converter(dataType):
 
 
 def _create_row(fields, values):
+    if len(values) > len(fields):
+        raise ValueError("Can not create %s by %s" % (fields, values))
```

Comment:
I'd like to improve this message a little, maybe "Can not create Row with fields %s, expected %d values but got %s" % (fields, len(fields), values)

Reply:
Thanks, improve done and move this check to `__call__` in `Row`. eb3f506